### ISSUE TITLE
Issue with tuples and unmanaged constraint

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#32103]
https://stackoverflow.com/questions/53992855/how-do-a-struct-containing-valuetuple-can-satisfy-unmanaged-constraint-but-valu

```cs
class Program
{
    static void DoSomething<T>() where T : unmanaged { }
    struct MyStruct {  public (int, int) Value; }
    static void Main(string[] args)
    {
        DoSomething<MyStruct>();    // → OK
        DoSomething<(int, int)>();  // → Shows compile-time error
    }
}
```

Reported by @johnkellyoxford

**Comment by johnkellyoxford** [comment]
Just an extra bit of info for anyone on this - using `sizeof` on the struct containing a tuple that contains only unmanaged types yields the correct expected size (8 for `(int, int)`, 12 for `(int, (int, int))`), and using the pointers works fine.

**Comment by RikkiGibson** [comment]
Right. Neither usage of `DoSomething` in the issue description is meant to be allowed in 7.3. Both usages will be allowed in 8.0. See dotnet/csharplang#1744.

**Comment by RikkiGibson** [comment]
Closing because the fix for this is now in master.

**Comment by jcouv** [comment]
Updated the milestone to 16.1p1 (the fix didn't make it into 16.0, I think).
